### ISSUE TITLE
crash after tournament finish

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#42]
This happened after a tournament successfully completed. I don't see line numbers, but `TempChat ` is calling `process` so that should be enough for me to figure out what's going on.

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', src/libcore/result.rs:1189:5
stack backtrace:
   0:     0x5557cf694e04 - backtrace::backtrace::libunwind::trace::hd843da7ff1c5f29e
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1:     0x5557cf694e04 - backtrace::backtrace::trace_unsynchronized::had390a3b5e3947d4
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2:     0x5557cf694e04 - std::sys_common::backtrace::_print_fmt::hf3539386bc544b27
                               at src/libstd/sys_common/backtrace.rs:84
   3:     0x5557cf694e04 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h929d1bf979a7bd14
                               at src/libstd/sys_common/backtrace.rs:61
   4:     0x5557cf6b750c - core::fmt::write::h41bac7029803ec62
                               at src/libcore/fmt/mod.rs:1025
   5:     0x5557cf68fc27 - std::io::Write::write_fmt::hd63968cca8037ffd
                               at src/libstd/io/mod.rs:1426
   6:     0x5557cf696f6e - std::sys_common::backtrace::_print::h83e9eed42de74378
                               at src/libstd/sys_common/backtrace.rs:65
   7:     0x5557cf696f6e - std::sys_common::backtrace::print::hb5610ce15b59e4ec
                               at src/libstd/sys_common/backtrace.rs:50
   8:     0x5557cf696f6e - std::panicking::default_hook::{{closure}}::h50a557df0a3289c9
                               at src/libstd/panicking.rs:193
   9:     0x5557cf696c61 - std::panicking::default_hook::he2237295883c0e7e
                               at src/libstd/panicking.rs:210
  10:     0x5557cf69764b - std::panicking::rust_panic_with_hook::hb938caf93064ba95
                               at src/libstd/panicking.rs:471
  11:     0x5557cf6971fe - rust_begin_unwind
                               at src/libstd/panicking.rs:375
  12:     0x5557cf6b363e - core::panicking::panic_fmt::h6c8ca41996834739
                               at src/libcore/panicking.rs:82
  13:     0x5557cf6b3737 - core::result::unwrap_failed::h2ac91e4397fb385b
                               at src/libcore/result.rs:1189
  14:     0x5557cf3b8a72 - <mb2::mb2::tournament::Tournament as mb2::mb2::Processor>::process::h95926da0e22f3d11
  15:     0x5557cf38683b - <mb2::mb2::nick_mapper::NickMapper as actix::handler::Handler<mb2::mb2::nick_mapper::TempChat>>::handle::h3153416b5291643e
  16:     0x5557cf3778f7 - <actix::address::envelope::SyncEnvelopeProxy<A,M> as actix::address::envelope::EnvelopeProxy>::handle::hbd7e25f8b58e2137
```

My guess is that `SendError` is due to mb2 trying to send a message to a disconnected user, probably my daughter Iris. IIRC, I have skimped a bit on some of the potential error conditions in part because the way we communicate via WebSockets is getting refactored.

**Comment by ctm** [comment]
I still haven't had a chance to look at this, but I did talk to Iris. She had quit her browser after she bust, but we didn't crash then. My guess is the code was smart enough to realize she was no longer in the tournament, but not smart enough to know that she was no longer in the starting pen.

Regardless, we shouldn't let a SendError crash us, because people will disconnect in the middle of a game.

**Comment by ctm** [comment]
It looks like this is the failure:
```
                self.to_crier
                    .send(Message::Player(player_id, message))
                    .unwrap();
```
to_crier` is a `Sender<Message>`. As such, the only reason for `Sender::send` to fail is a disconnect, which is not what I thought was going on, but does make sense and should be easy to reproduce and fix.

**Comment by ctm** [comment]
Yup. It's an issue with how we're doing messages from IRC tables. Apparently enough of the table is left behind after a finish that someone on, for example, `#table-1` chatting will get mb2's attention and result in `mb2` finding the tournament corresponding to `table-1` and then trying to get it to send chat, only to find that the `to_crier` is disconnected.

There is more than one issue here. The biggest, IMO, is that we don't ever want to be in a situation where a `to_crier` is disconnected. If a tournament is finished, we want the `Tournament` that was running it to go away entirely, so there's no `to_crier` field to attempt to send a message on anyway.

FWIW, I have not been able to reproduce this with just the web client, which explains why I haven't seen the bug myself; once I implemented chat I switched almost all my testing to be with the web client.
